interface/modules/custom_modules/oe-module-weno/src/Services/LogProperties.php

```diff
@@ -167,9 +167,20 @@ public function logSync()
             if ($isError['is_error']) {
                 $error = $isError['messageText'];
                 error_log('Prescription download failed: ' . errorLogEscape($error));
-                $wenoLog->insertWenoLog("prescription", "Invalid Prescriber Credentials Hint:see previous errors.");
                 EventAuditLogger::instance()->newEvent("prescriptions_log", $_SESSION['authUser'], $_SESSION['authProvider'], 0, ($error));
-                die(js_escape($error));
+                // if background task then return false
+                if ($tasked == 'background') {
+                    $wenoLog->insertWenoLog("prescription", "Invalid Prescriber Credentials Background Service.");
+                    return false;
+                }
+                if ($tasked == 'downloadLog') {
+                    $wenoLog->insertWenoLog("prescription", "Invalid Prescriber Credentials User Download.");
+                    echo(js_escape($error));
```

Comment:
would use text() instead of js_escape()

Reply:
Needs to be js escaped as it is a return to ajax downloadPresLog().
It gets escaped with 
```
let errorMsgSpan = document.getElementById('error-msg');
errorMsgSpan.textContent = jsText(data);
```

Reply:
Ajax server scripts still need to do html escaping (in case a bad actor calls them directly from browser) with text/attr. js_escape won't prevent xss on these if bad actor calls the ajax server script directly from browser. Then will also need to re-escape it when run it on the ajax client script when bring it into html element (like you are doing). Escaping is fun stuff :)

Reply:
not going to do!